### ISSUE TITLE
Add support for WebExtension dictionaries

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-linter#1984]
Part of https://github.com/mozilla/addons-server/issues/6739

The first part of the schema support for that lands with https://github.com/mozilla/addons-linter/pull/1983/

**Comment by EnTeQuAk** [comment]
Some more information on how to tackle this:

Add a new `validateDictionary` function in `schema/validator.js` similarly to https://github.com/mozilla/addons-linter/blob/master/src/schema/validator.js#L84-L114
This needs to select the correct parts in the schema. I think something like

```javascript

const _validateDictionary = validator.compile({
  ...schemaObject,
  id: 'dictionary-manifest',
  $ref: '#/types/WebExtensionDictionaryManifest'
});
```

will probably work, plus all that fuzz regarding `additionalProperties` if needed.

This can then be used in the `parsers/manifestjson.json` instead of `validateAddon`.

To add support for separate file types (`.dic` or stuff like that) you can add new scanners for them and link them in `linter.js:Linter.getScanner` which will run their scanner and allows you to validate them separately.

Let me know if that helps you implementing it.

**Comment by diox** [comment]
Dictionaries are a webextension with a manifest containing the `dictionaries` property, which is an object of the form: `{ "localename": "/path/to/locale.dic"}`. Inside the extension, the `/path/to/locale.dic` path must exist, whatever it is, must match what's specified in the `dictionaries` object, and there must be a corresponding `.aff` file placed in the same directory.

Only single-locale dictionaries are supported at the moment, so a manifest with `"dictionaries": {"locale1": "/some/file.dic", "locale2": "/some/other/file.dic"}` should raise an error.

You can use a legacy dictionary to get real `.dic` and `.aff` files, but their contents are not validated anyway (they weren't before with the legacy validator either).

**Comment by AlexandraMoga** [comment]
Verified on AMO -dev with FF61, Win10x64

The linter will read the `dictionaries` property from the manifest and will apply validation rules specific to this add-on type.

To prove this works, I have tried to trigger several specific errors during validation:
- supplying an invalid property (i.e. csp, permissions etc)
- adding multiple languages
- removing a dictionary file - `.dic` or `.aff`
- including a `strict-max-version` restriction 

Validation errors generated:
![image](https://user-images.githubusercontent.com/31961530/43766673-c423dcf8-9a3b-11e8-8090-23f5b67fe0e0.png)

Link to validation results: https://addons-dev.allizom.org/en-US/developers/upload/fe7226a84a114bf780baf5b4f187d9d4

@diox I've noticed that the linter will not detect invalid languages, something like:

```json
"dictionaries": {
    "foo": "/dictionaries/foo.dic"
}
```
I'm not sure if this is something worth supporting in the future though.

**Comment by diox** [comment]
Yeah, I didn't want to have a whitelist of valid locales in the linter itself, we could want completely arbitrary ones...
